### DISCUSSION TITLE
How is the memory handled?

### DISCUSSION THREAD

**Discussion by sagarspatil** [ztjhz/BetterChatGPT#140] [General]
Hello,

How does the memory feature work? I know the data is stored locally, but do you embed the conversations and share them with the model? How can it refer to a conversation say that is beyond 8000 tokens? (In case of GPT-4) or will it lose context?

**Comment by ztjhz** [comment]
It shares the entire conversation to the model. If it exceeds 8000 tokens (for GPT-4), the conversation will be truncated to less than 8000 tokens.

**Reply by jonmatthis** [reply]
I was thinking about integrating some of the [LangChain memory management](https://python.langchain.com/en/latest/modules/memory.html)  methods into this (lovely lovely) project

Does that align with your roadmap? If so, I'd love make a PR back into this branch when I've got it working!

**Reply by jonmatthis** [reply]
LangChain is Python based, so integrating it might be a bit tricky, but I think it would add a lot of potential applications to this project. 

This project implements a nice combo of ReactJS frontend code with Python based backend code, with communication managed through a `docker-compose.yml` that makes containers for each one - https://github.com/logspace-ai/langflow

**Reply by jonmatthis** [reply]
Anyway, let me know if y'all have any plans for handling Memory in a more sophisticated way - if not, I'll just poke around and let you know if I figure out anything useful :) 

It might take the form of a PR into your repo, or maybe just an integration with some other codebase that I will make on own (if it looks like it will be hard to do with out making a lot of deep changes to  your repo)

**Reply by ztjhz** [reply]
We do plan to handle memory in a more sophisticated way!
